Util/openl-maven-plugin/src/org/openl/rules/maven/TestMojo.java

```diff
@@ -213,16 +218,24 @@ private Summary executeAllAtOnce(String testSourcePath,
                 .setExternalParameters(externalParameters)
                 .build();
 
+            validateOpenApi(factory);
+
             CompiledOpenClass openLRules = factory.getCompiledOpenClass();
             return executeTests(openLRules);
         } finally {
             OpenClassUtil.releaseClassLoader(classLoader);
         }
     }
 
+    private void validateOpenApi(SimpleProjectEngineFactory<?> factory) throws ProjectResolvingException, RulesInstantiationException {
+        OpenApiProjectValidator openApiProjectValidator = new OpenApiProjectValidator();
+        openApiProjectValidator.validate(factory.getProjectDescriptor(), factory.getRulesInstantiationStrategy());
+    }
+
     private Summary executeModuleByModule(String testSourcePath,
             String mainSourcePath,
             boolean hasDependencies) throws MalformedURLException, ProjectResolvingException {
+        log.warn(singleModuleMode ? "singleModuleMode " : "maxModulesInMemory" + "variable was used. OpenApi validation was skipped.");

```

Comment:
we can't validate project in this method because we compile it module by module so we can't validate it fully and inform that if this mode was chosen we're don't verify open api